internal/backend/sessions/sessioncalls/call.go

```diff
@@ -77,7 +81,41 @@ func (cs *calls) invoke(ctx context.Context, callv sdktypes.Value, args []sdktyp
 		return sdktypes.InvalidSessionCallAttemptResult, fmt.Errorf("executor not found: %q", xid)
 	}
 
-	return sdktypes.NewSessionCallAttemptResult(caller.Call(ctx, callv, args, kwargs)), nil
+	if timeout != 0 {
+		var cancel context.CancelFunc
+		ctx, cancel = context.WithTimeout(ctx, timeout)
+		defer cancel()
+
+		if !activity.IsActivity(ctx) {
+			if wctx := sessioncontext.GetWorkflowContext(ctx); wctx != nil {
+				wctx, cancel := workflow.WithCancel(wctx)
+				ctx = sessioncontext.WithWorkflowContext(ctx, wctx)
+				go func() {
+					<-ctx.Done()
+					cancel()
+				}()
+			}
+		}
+	}
+
+	v, err := caller.Call(ctx, callv, args, kwargs)
+
+	if err != nil && errors.Is(err, workflow.ErrCanceled) && errors.Is(ctx.Err(), context.DeadlineExceeded) {
```

Comment:
Use this condition in line 107 and get rid of this section?

Reply:
it's comparing two different things, err and ctx.Err, so not so simple